File: TASC-Intern-Batch9/pharmacy-backend-microservice/cart-service/src/main/java/com/pharmacy_backend/cart_service/service/CartService.java
package com.pharmacy_backend.cart_service.service;

import com.pharmacy_backend.cart_service.dto.request.CartItemRequest;
import com.pharmacy_backend.cart_service.dto.response.CartItemResponse;
import com.pharmacy_backend.cart_service.dto.response.CartResponse;
import com.pharmacy_backend.cart_service.entity.Cart;
import com.pharmacy_backend.common.dto.response.ApiResponse;

import java.util.List;

public interface CartService {
    ApiResponse<CartResponse> getCart();
    ApiResponse<CartItemResponse> addItemToCart(CartItemRequest request);
    ApiResponse<CartItemResponse> updateItemQuantity(Long itemId, Integer quantity);
    ApiResponse<Void> removeItemFromCart(Long itemId);
    ApiResponse<Void> clearCart();
    ApiResponse<CartItemResponse> changeItemSelection(Long itemId, Boolean status);
    ApiResponse<List<CartItemResponse>> selectAllItems(Boolean status);
    ApiResponse<List<CartResponse>> getCartItemsForCheckout();
    ApiResponse<Long> getTotalItemsInCart();
    Cart createCart(Long userId);
}

File: TASC-Intern-Batch9/fe/fe-client/src/api/cartService.ts
import axiosClient from './axiosClient';
import type { ApiResponse } from '../types';
import type { CartItemRequest, CartItemResponse, CartResponse } from '../types/cart.types';

const cartService = {
  // Lấy toàn bộ giỏ hàng
  getCart: () => {
    return axiosClient.get<ApiResponse<CartResponse>>('/carts');
  },

  // Lấy tổng số lượng item (để hiện badge trên header)
  getTotalItems: () => {
    return axiosClient.get<ApiResponse<number>>('/carts/item/totalItems');
  },

  // Thêm sản phẩm
  addItemToCart: (data: CartItemRequest) => {
    return axiosClient.post<ApiResponse<CartItemResponse>>('/carts', data);
  },

  // Cập nhật số lượng (PUT /carts/item/{itemId}?quantity=...)
  updateItemQuantity: (itemId: number, quantity: number) => {
    return axiosClient.put<ApiResponse<CartItemResponse>>(`/carts/item/${itemId}`, null, {
      params: { quantity }
    });
  },

  // Chọn/Bỏ chọn 1 sản phẩm (PATCH /carts/item/status/{itemId}?selected=...)
  updateItemStatus: (itemId: number, status: boolean) => {
    return axiosClient.patch<ApiResponse<CartItemResponse>>(`/carts/item/status/${itemId}`, null, {
      params: { selected: status }
    });
  },

  // Chọn/Bỏ chọn tất cả (PUT /carts/item/status/all?selected=...)
  selectAllItems: (status: boolean) => {
    return axiosClient.put<ApiResponse<CartItemResponse[]>>('/carts/item/status/all', null, {
      params: { selected: status }
    });
  },

  // Xóa 1 sản phẩm
  removeItemFromCart: (itemId: number) => {
    return axiosClient.delete<ApiResponse<void>>(`/carts/item/${itemId}`);
  },

  // Xóa toàn bộ giỏ hàng
  clearCart: () => {
    return axiosClient.delete<ApiResponse<void>>('/carts');
  }
};

export default cartService;